cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -388,19 +388,10 @@ func (g *gitHubRepository) downloadFilesFromRelease(release *github.RepositoryRe
 			return false, nil
 		}
 		if redirect != "" {
-			req, err := http.NewRequestWithContext(ctx, http.MethodGet, redirect, http.NoBody)
-			if err != nil {
-				retryError = errors.Wrapf(err, "failed to download file %q from %q release via redirect location %q: failed to create request", *release.TagName, fileName, redirect)
-				return false, nil
-			}
-
-			response, err := http.DefaultClient.Do(req) //nolint:bodyclose // (NB: The reader is actually closed in a defer)
-			if err != nil {
-				retryError = errors.Wrapf(err, "failed to download file %q from %q release via redirect location %q", *release.TagName, fileName, redirect)
-				return false, nil
-			}
-			reader = response.Body
+			retryError = errors.New("unexpected redirect while downloading the release asset")
```

Comment:
Sorry for another nit. Would it be possible to tell folks which release asset we tried to download?

Reply:
also, if possible, while at it please add a note about why we are handling this as a error
 ```suggestion
			// NOTE: DownloadReleaseAsset should not return a redirect address when used with the DefaultClient
			retryError = errors.New("unexpected redirect while downloading the release asset")
```

Reply:
Added the comment.

@sbueringer I don't think we need to add that. This is already there in wraping above this function:

```
❯ go run ./cmd/clusterctl init
Fetching providers
Error: failed to get provider components for the "cluster-api" provider: failed to read "core-components.yaml" from provider's repository "cluster-api": failed to download files from GitHub release v1.3.2: unexpected redirect while downloading the release asset
```

Reply:
(happy to add it)